core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -398,9 +395,6 @@ final class Request[F[_]] private (
 
   def remoteAddr: Option[IpAddress] = remote.map(_.host)
 
-  def remoteHost(implicit F: Monad[F], dns: Dns[F]): F[Option[Hostname]] =
-    OptionT.fromOption(remote.map(_.host)).flatMapF(dns.reverseOption).value
```

Comment:
Can this be written like this?
```scala
def remoteHost[F2[x] >: F[x]](implicit F: Monad[F2], dns: Dns[F2]): F2[Option[Hostname]]
```

Reply:
Yep absolutely. Probably better to do this one that way.

Reply:
Actually, can't many several of the methods currently in invariant ops be written like this?